indra/llcharacter/llik.cpp

```diff
@@ -856,11 +867,22 @@ void LLIK::DoubleLimitedHinge::dumpConfig() const
 }
 #endif
 
-LLIK::Joint::Joint(S16 id, const LLVector3& local_pos, const LLVector3& bone)
-    :   mDefaultLocalPos(local_pos), mBone(bone), mID(id)
+LLIK::Joint::Joint(LLJoint* info_ptr) : mInfoPtr(info_ptr)
 {
-    mLocalPos = mDefaultLocalPos;
+    assert(info_ptr != nullptr);
+    mID = mInfoPtr->getJointNum();
+    resetFromInfo();
+}
+
+void LLIK::Joint::resetFromInfo()
+{
+    LLVector3 scale = mInfoPtr->getScale();
+    mLocalPos = mInfoPtr->getPosition().scaledVec(scale);
+    mBone = mInfoPtr->getEnd().scaledVec(scale);
     mLocalPosLength = mLocalPos.length();
+    // This is Correct: we do NOT store info scale in mLocalScale.
+    // mLocalScale represents Puppetry's tweak on  top of whatever is set in the info.
+    mLocalScale.set(1.0f, 1.0f, 1.0f);
 }
```

Comment:
`LLIK::Joint::resetFromInfo()` replaces the old `LLIK::Joint::reconfigure()` method below.